PWGGA/GammaConvBase/AliConvEventCuts.cxx

```diff
@@ -6278,93 +6268,102 @@ void AliConvEventCuts::GetNotRejectedParticles(Int_t rejection, TList *HeaderLis
       if (fDebugLevel > 0 ) cout << i << "\t" << GeneratorName.Data() << endl;
       for(Int_t j = 0; j<HeaderList->GetEntries();j++){
         TString GeneratorInList = ((TObjString*)HeaderList->At(j))->GetString();
-        if(GeneratorInList.Contains(GeneratorName) ){
-          if (GeneratorInList.Contains("PARAM") || GeneratorInList.CompareTo("BOX") == 0 ){
-            if(fMCEvent){
-              if (fPeriodEnum == kLHC14a1b || fPeriodEnum == kLHC14a1c ){
-                if (fMCEvent->Particle(firstindex)->GetPdgCode() == fAddedSignalPDGCode ) {
-                  if (fDebugLevel > 0 ) cout << "produced " << gh->NProduced() << " with box generator" << endl;
-                  if (gh->NProduced() > 10 && fMCEvent->Particle(firstindex+10)->GetPdgCode() == fAddedSignalPDGCode && GeneratorInList.CompareTo("BOX") == 0){
-                    if (fDebugLevel > 0 ) cout << "one of them was a pi0 or eta" <<  endl;
-                    fNotRejectedStart[number] = firstindex;
-                    fNotRejectedEnd[number] = lastindex;
-                    fGeneratorNames[number] = GeneratorName;
-                    number++;
-                    if (fDebugLevel > 0 ) cout << "Number of particles produced for: " << i << "\t" << GeneratorName.Data() << "\t" << lastindex-firstindex+1 << endl;
-                    continue;
-                  } else if (gh->NProduced() == 3 && GeneratorInList.Contains("PARAM_EMC") && (i == 3 || i == 5) ){
-                    fNotRejectedStart[number] = firstindex;
-                    fNotRejectedEnd[number] = lastindex;
-                    fGeneratorNames[number] = GeneratorName;
-                    number++;
-                    if (fDebugLevel > 0 ) cout << "Number of particles produced for: " << i << "\t" << GeneratorName.Data() << "\t" << lastindex-firstindex+1 << endl;
-                    continue;
-                  } else if (gh->NProduced() > 2 && GeneratorInList.Contains("PARAM_PHOS") && (i == 4 || i == 6) ){
-                    fNotRejectedStart[number] = firstindex;
-                    fNotRejectedEnd[number] = lastindex;
-                    fGeneratorNames[number] = GeneratorName;
-                    number++;
-                    if (fDebugLevel > 0 ) cout << "Number of particles produced for: " << i << "\t" << GeneratorName.Data() << "\t" << lastindex-firstindex+1 << endl;
-                    continue;
+        if (fPeriodEnum==kLHC20g10){
```

Comment:
same here. Only the if in line 6271 gets added, everything else that was here goes into its else